server/src-lib/Hasura/GraphQL/Execute/LiveQuery/State.hs

```diff
@@ -107,16 +116,22 @@ addLiveQuery lqState plan onResultAction = do
     newPoller = Poller <$> TMap.new <*> STM.newEmptyTMVar
 
 removeLiveQuery
-  :: LiveQueriesState
+  :: L.Logger L.Hasura
+  -> LiveQueriesState
   -- the query and the associated operation
   -> LiveQueryId
   -> IO ()
-removeLiveQuery lqState (LiveQueryId handlerId cohortId sinkId) = do
-  threadRef <- STM.atomically $ do
+removeLiveQuery logger lqState lqId@(LiveQueryId handlerId cohortId sinkId) = do
+  threadRefM <- STM.atomically $ do
     detM <- getQueryDet
     fmap join $ forM detM $ \(Poller cohorts ioState, cohort) ->
       cleanHandlerC cohorts ioState cohort
-  traverse_ A.cancel threadRef
+  case threadRefM of
+    Just threadRef -> Immortal.stop threadRef
+    -- This would seem to imply addLiveQuery broke, or a bug elsewhere:
+    Nothing -> L.unLogger logger $ L.UnstructuredLog L.LevelError $ fromString $
+      "In removeLiveQuery no worker thread installed. Please report this as a bug: "<>
+      show lqId
```

Comment:
Might as well log a more structured message instead of using the show instance on `LiveQueryId`?

Reply:
I know it's a bit ugly, but it didn't seem worth the effort for an unexpected/internal error. Let me know if you strongly object!